lib/ansible/modules/cloud/vmware/vmware_host_tcpip_stacks.py

```diff
@@ -0,0 +1,473 @@
+            vmotion_congestion_algorithm = self.params['vmotion'].get('congestion_algorithm')
+            vmotion_num_connections = self.params['vmotion'].get('num_connections')
+        if self.params['vxlan']:
+            vxlan_vmkernel_gateway = self.params['vxlan'].get('vmkernel_gateway')
+            vxlan_congestion_algorithm = self.params['vxlan'].get('congestion_algorithm')
+            vxlan_num_connections = self.params['vxlan'].get('num_connections')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            changed_list = []
+            results['result'][host.name] = dict(changed='', msg='')
+
+            host_netstack_config = host.config.network.netStackInstance
+
+            changed_instances_list = []
+            for instance in host_netstack_config:
+                netstack_spec = vim.host.NetworkConfig.NetStackSpec()
+                netstack_spec.netStackInstance = vim.host.NetStackInstance()
+                changed_instance = False
+                # Check default TCP/IP stack
+                if instance.key == 'defaultTcpipStack':
+                    results['result'][host.name]['default_vmkernel_gateway'] = default_vmkernel_gateway
+                    results['result'][host.name]['default_num_connections'] = default_num_connections
+                    results['result'][host.name]['default_congestion_algorithm'] = default_congestion_algorithm
+                    (netstack_spec,
+                     changed_instance,
+                     vmkernel_gateway_previous,
+                     num_connections_previous,
+                     congestion_algorithm_previous) = self.create_netstack_spec(
+                         instance,
+                         'defaultTcpipStack',
+                         default_vmkernel_gateway,
+                         default_num_connections,
+                         default_congestion_algorithm)
+                    if changed_instance:
+                        changed = True
+                        changed_list.append("default TCP/IP stack")
+                        # vmkernel_gateway_previous can be Null
+                        if default_vmkernel_gateway:
+                            results['result'][host.name]['default_vmkernel_gateway_previous'] = vmkernel_gateway_previous
+                        if num_connections_previous:
+                            results['result'][host.name]['default_num_connections_previous'] = num_connections_previous
+                        if congestion_algorithm_previous:
+                            results['result'][host.name]['default_congestion_algorithm_previous'] = \
+                                congestion_algorithm_previous
+                        changed_instances_list.append(netstack_spec)
+                # Check Provisioning TCP/IP stack
+                if instance.key == 'vSphereProvisioning':
```

Comment:
This could be refactored into a method, since the logic is identical.  Same for the next two large blocks.

Reply:
Agreed